plugin/release/src/main/kotlin/net/twisterrob/gradle/android/AndroidMinificationPlugin.kt

```diff
@@ -93,25 +93,40 @@ class AndroidMinificationPlugin : BasePlugin() {
 
 		lintTasksDependOnProguardRulesTask(extractMinificationRules)
 		project.afterEvaluate {
-			android.variants.all { variant ->
-				val proguardTask = proguardTaskClass?.let { project.findMinificationTaskFor(variant, it) }
-				val r8Task = project.findMinificationTaskFor(variant, R8Task::class.java)
-				val obfuscationTask = proguardTask ?: r8Task
-				if (obfuscationTask != null) {
-					obfuscationTask.dependsOn(extractMinificationRules)
-					if (variant.flavorName == "") {
-						val generateMinificationRulesTask = createGenerateMinificationRulesTask(
-							variant,
-							generatedProguardRulesFile,
-							proguardTask == obfuscationTask
-						)
-						lintTasksDependOnProguardRulesTask(generateMinificationRulesTask)
-						obfuscationTask.dependsOn(generateMinificationRulesTask)
-					} else {
-						// Cannot do this simply because AGP doesn't provide a DSL surface for adding proguard rules for variants.
-						// It's possible to add for buildType and flavors, but since the mapping file is for variants,
-						// the generate... task would not be able to create a distinct file.
-						obfuscationTask.logger.warn("This project uses flavors, it's not possible to generate variant based rules for ${variant.name}.")
+			android.variants.configureEach { variant ->
+				val isFlavorless = variant.flavorName == ""
+				if (!isFlavorless) {
+					// Cannot do this simply because AGP doesn't provide a DSL surface for adding proguard rules for variants.
+					// It's possible to add for buildType and flavors, but since the mapping file is for variants,
+					// the generate... task would not be able to create a distinct file.
+					project.logger.warn("This project uses flavors, it's not possible to generate variant based rules for ${variant.name}.")
+				}
```

Comment:
validate if this still stands, maybe generate one with maybeCreate and then use that?

Reply:
the `printconfiguration` depends on the variant, so registering multiple would not work.